app/models/spree/sale_price.rb

```diff
@@ -1,6 +1,11 @@
 module Spree
   class SalePrice < ActiveRecord::Base
-    include Spree::SoftDeletable
+
+    if ::Spree.solidus_gem_version >= Gem::Version.new('2.11')
+      include Spree::SoftDeletable
+    else
+      include SolidusSalePrices::SoftDelete
```

Comment:
I don't see the point in having this in a concern and including it in only one place. This code is temporary until all supported versions of Solidus have `Spree::SoftDeletable`, so let's make this code easier to delete and just inline it.

Reply:
I see your point on this and looking back at it now, I reimplemented the same functionality of Spree::SoftDeleteable so should we even include the conditional at all? I am thinking of just leaving
```
include SolidusSalePrices::SoftDelete
```
and open an issue to replace that with Spree::SoftDeleteable once 2.10 is no longer supported. What do you think @jarednorman?

Reply:
I think there's something subtle that I missed on my first review: `SolidusSalePrices::SoftDelete` uses paranoia, but `Spree::SoftDeletable` does not. This means that using the former means you can call `.destroy` on the record and cause it to get soft-deleted, whereas with the latter doing so will cause the record to actually be dropped from the database.

If we care about this difference, I'd say we maintain the current functionality and leave a note in the codebase regarding how we make the breaking change in the future to move to the soft-deletion functionality that comes with Solidus:

```ruby
# The following code enables soft-deletion. In Solidus v2.11+ there is a mixin
# that handles soft-deletion consistently across all Solidus records that need
# it. Once we no longer support v2.10, we can remove the dependency on paranoia
# and replace these lines with:
#
#     include Spree::SoftDeletable
#
# However, this will be a breaking change as it will change the behaviour of
# calling `destroy` on these records.
acts_as_paranoid
include Discard::Model
self.discard_column = :deleted_at
```